### PR TITLE
Changes how tgui handles static data

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
So tgui sends two types of data- standard and static. For optimization purposes you can send data as static data - but there's a problem if the user updates the screen. Refreshing the screen does not always send static data as it's on a cooldown.

Good UIs will have the error case handled where there is no static data retrieved.
Most will just not render anything, like vending machines, so forth.
Almost all UIs will try to cover for this case by casting objects as arrays so they can execute .map on them.

Here's an example of a UI which uses static data. If I tap F5 it will eventually hit the refresh cooldown and not send anything at all. This isn't a "bug in code" per se, it's just how it works and I disagree with that design. You can see that it has the error case handled for no static data already:
<details>
<summary><b>VIEW</b></summary>

dreamseeker_yHg0ouLX4S
</details>

I've changed it to where DM will wait for the cooldown to send new static data over. I've also added a canary of sorts to the internals of TGUI that detects when it's waiting on this data refresh. This should outright prevent any bluescreens that occur if/when someone tries to act on data that isn't sent over due to a refresh. No more .length of undefined, .includes of undefined etc. And it gives a visual cue while doing so:
<details>
<summary><b>VIEW</b></summary>

dreamseeker_AUQt34qDPQ
</details>

oh and of course i messed with some window sizing and stuff elsewhere

## Why It's Good For The Game
Uis can safely rely on static data
Since the timer is built into ```send_full_update``` now, other cases where UIs use it no longer bypass the refresh cooldown
The user will understand the ui isn't "broken" if it's just waiting

## Changelog

:cl:
fix: TGUI now alerts the user when it is waiting on a refresh cooldown. In many cases, this will prevent screens from breaking that rely on static data.
/:cl:

### CODE REVIEW

File: code/modules/tgui/tgui.dm [hunk 1]
```diff
@@ -305,8 +309,13 @@
 	switch(type)
 		if("ready")
 			// Send a full update when the user manually refreshes the UI
-			if (initialized && COOLDOWN_FINISHED(src, refresh_cooldown))
-				send_full_update()
+			if (initialized)
+				if(!COOLDOWN_FINISHED(src, refresh_cooldown))
+					if(!refreshing)
+						addtimer(CALLBACK(src, .proc/send_full_update), TGUI_REFRESH_FULL_UPDATE_COOLDOWN)
```
Comment (optimumtact) [inline code/modules/tgui/tgui.dm:315]: make sure this is a unique timer so it can't be queued or spammed

Comment (jlsnow301) [reply code/modules/tgui/tgui.dm:315]: indeed - that's what i'm using the ```refreshing``` var for. the only way to flick off refreshing is for send_full_update to fire on the window.

if there is a better way to do this, let me know, i'm not very familiar with timers

Comment (stylemistake) [reply code/modules/tgui/tgui.dm:315]: It's a confusing design, but I don't know what to suggest.

Comment (jlsnow301) [reply code/modules/tgui/tgui.dm]: i tried writing it differently. dont know how/why i let it get to 3 nested ifs

**Review by stylemistake — APPROVED** [review]

**Pull request merged**